### PR TITLE
core(offline-start-url): better failure messages

### PR DESCRIPTION
**Pull request by connorjclark**
Resolves an issue brought up in #4261

1. Add `url` to the `StartUrl` artifact. I know it's already in the manifest artifact,  but it's awkward to access
1. if non 200 but there was indeed a SW handling the request, form a better message
1. hack off the fragment when sniffing for the start url request (was causing the "timeout" explanation to happen for `/#test`)
1. include start url in timeout explanation

Open question: warning or explanation (ignoring the known poor UX)?

base:
image
explanation:
image
warning:
image

### CODE REVIEW

File: lighthouse-core/gather/gatherers/start-url.js [hunk 1]
```diff
@@ -37,9 +37,15 @@ class StartUrl extends Gatherer {
...
-      return await this._attemptStartURLFetch(passContext.driver, startUrlInfo.startUrl);
+      const statusAndExplanation =
+        await this._attemptStartURLFetch(passContext.driver, startUrlInfo.startUrl);
+      return {url: startUrlInfo.startUrl, ...statusAndExplanation};
     } catch (err) {
-      return {statusCode: -1, explanation: 'Error while fetching start_url via service worker.'};
+      return {
+        url: startUrlInfo.startUrl,
+        statusCode: -1,
+        explanation: 'Error while fetching start_url via service worker.',
```
Comment (connorjclark) [inline lighthouse-core/gather/gatherers/start-url.js]: this whole file needs some i18n love @exterkamp

File: lighthouse-core/gather/gatherers/start-url.js [hunk 2]
```diff
@@ -75,19 +81,25 @@ class StartUrl extends Gatherer {
...
+        () => resolve({
+          statusCode: -1,
+          explanation: `Timed out waiting for start_url (${startUrl}) to respond.`,
+        }),
         3000
       )
     );
 
+    const startUrlObject = new URL(startUrl);
+    const startUrlNoHash = startUrlObject.origin + startUrlObject.pathname;
```
Comment (paulirish) [inline lighthouse-core/gather/gatherers/start-url.js]: reconstructing URLs with concatenation is super frightening to me.

```suggestion
    startUrlObject.hash = '';
    const startUrlNoHash = startUrlObject.toString();
```

File: lighthouse-core/audits/offline-start-url.js [hunk 3]
```diff
@@ -56,6 +56,19 @@ class OfflineStartUrl extends Audit {
 
     const hasOfflineStartUrl = artifacts.StartUrl.statusCode === 200;
 
+    // The StartUrl artifact provides no explanation if a response was received from a service
+    // worker with a non-200 status code. In that case, this audit provides its own explanation.
+    // In all other cases it defers to the artifact explanation.
+    const explanation = artifacts.StartUrl.explanation;
+    // if (!explanation && artifacts.StartUrl.statusCode !== -1 && !hasOfflineStartUrl) {
```
Comment (paulirish) [inline lighthouse-core/audits/offline-start-url.js]: presumably you're gonna nuke these lines?

Comment (connorjclark) [reply lighthouse-core/audits/offline-start-url.js]: based on feedback re: warning or explanation, yes

File: lighthouse-core/gather/gatherers/start-url.js (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/gather/gatherers/start-url.js]: ya this smarrtyer

File: lighthouse-core/audits/offline-start-url.js (hunk 3 above)
Comment (patrickhulce) [reply lighthouse-core/audits/offline-start-url.js]: explanation makes more sense IMO

File: lighthouse-core/gather/gatherers/start-url.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/start-url.js]: I think all our gatherers have this problem :)

File: lighthouse-core/gather/gatherers/start-url.js [hunk 4]
```diff
@@ -75,19 +81,26 @@ class StartUrl extends Gatherer {
...
+        () => resolve({
+          statusCode: -1,
+          explanation: `Timed out waiting for start_url (${startUrl}) to respond.`,
+        }),
         3000
       )
     );
 
+    const startUrlObject = new URL(startUrl);
+    startUrlObject.hash = '';
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/start-url.js]: I thought we had a utility function on `URL` for this

**Pull request merged**